plugin/nomad/setup.go

```diff
@@ -0,0 +1,117 @@
+package nomad
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/coredns/caddy"
+	"github.com/coredns/coredns/core/dnsserver"
+	"github.com/coredns/coredns/plugin"
+
+	nomad "github.com/hashicorp/nomad/api"
+)
+
+// init registers this plugin.
+func init() { plugin.Register(pluginName, setup) }
+
+// setup is the function that gets called when the config parser see the token "nomad". Setup is responsible
+// for parsing any extra options the nomad plugin may have. The first token this function sees is "nomad".
+func setup(c *caddy.Controller) error {
+	n := &Nomad{
+		ttl:     uint32(defaultTTL),
+		clients: make([]*nomad.Client, 0),
+		current: -1,
+	}
+	if err := parse(c, n); err != nil {
+		return plugin.Error("nomad", err)
+	}
+
+	c.OnStartup(func() error {
+		var err error
+		for idx, client := range n.clients {
+			_, err := client.Agent().Self()
+			if err == nil {
+				n.current = idx
+				return nil
+			}
+		}
+		return err
+	})
+
+	dnsserver.GetConfig(c).AddPlugin(func(next plugin.Handler) plugin.Handler {
+		n.Next = next
+		return n
+	})
+
+	return nil
+}
+
+func parse(c *caddy.Controller, n *Nomad) error {
+	var token string
+	addresses := []string{} // Multiple addresses are stored here
+
+	for c.Next() {
+		for c.NextBlock() {
+			selector := strings.ToLower(c.Val())
+
+			switch selector {
+			case "address":
+				addresses = append(addresses, c.RemainingArgs()[0])
```

Comment:
To not just append the first address, shouldn't this be:

```suggestion
				addresses = append(addresses, c.RemainingArgs()...)
```

Maybe add tests for the setup? You can check other `setup_test.go` implementations for reference. Should be fairly straightforward.